tests/unittests/test_sshutil.py

```diff
@@ -326,4 +366,76 @@ def test_not_modified(self):
         m_write_file.assert_not_called()
 
 
+class TestMultipleSshAuthorizedKeysFile(test_helpers.CiTestCase):
+    username = 'foouser'
+    dist = MyBaseDistro()
+    dist.create_user(username)
```

Comment:
I believe you can skip this setup, the only use of user is in path construction, in ssh_util the 'user_ssh_info' call returns the path to users .ssh directory and the user name in a tuple.  You should be able to mock that function.

@mock.patch('cloudinit.ssh_util.users_ssh_info', m_ssh_info)

And then inject the values you want to mock, something like this (untested but should give you the right idea)

```
  from collections import namedtuple
  
  pwdnam = namedtuple('pwdnam', ['pw_dir'])

  class TestMultipleSsh(CiTestCase):
      @mock.patch('cloudinit.ssh_util.users_ssh_info')
       def test_multiple_ssh(self, m_ssh_info):
           userdir = self.tmp_dir()
           ssh_dir = os.path.join(userdir, '.ssh')
           os.makedirs(ssh_dir)
           m_ssh_info.return_value = (ssh_dir, pwdnam(userdir))
           ....
           ssh_util.extract_authorized_keys()
                  
```
  
This way you can drop the distro class.

Reply:
Actually I need the user creation because I need to pass it to extract_authorized_keys(), this function will check the username and extract and store the the keys from its private file.

Reply:
I don't want to change a function and cause a refactoring across the whole project because of a test :-)

Reply:
You' not really refactoring across the whole project.
In the change in my branch, there are 3 functions changed:

 * parse_authorized_keys: goes from taking a single filename to a list of filenames.  this had only 1 caller.
 * extract_authorized_keys: adds a kwarg for default value of ssh_cfg_file to read.  This had 2 callers.
 * render_authorizedkeysfile_paths: added function (split out of extract_authorized_keys)